<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN"
    "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
<head>
<meta http-equiv="Content-Type" content="application/xhtml+xml; charset=UTF-8" />
<meta name="generator" content="AsciiDoc 8.6.9" />
<title>TransDecoder (Find Coding Regions Within Transcripts)</title>
<style type="text/css">
/* Shared CSS for AsciiDoc xhtml11 and html5 backends */

/* Default font. */
body {
  font-family: Georgia,serif;
}

/* Title font. */
h1, h2, h3, h4, h5, h6,
div.title, caption.title,
thead, p.table.header,
#toctitle,
#author, #revnumber, #revdate, #revremark,
#footer {
  font-family: Arial,Helvetica,sans-serif;
}

body {
  margin: 1em 5% 1em 5%;
}

a {
  color: blue;
  text-decoration: underline;
}
a:visited {
  color: fuchsia;
}

em {
  font-style: italic;
  color: navy;
}

strong {
  font-weight: bold;
  color: #083194;
}

h1, h2, h3, h4, h5, h6 {
  color: #527bbd;
  margin-top: 1.2em;
  margin-bottom: 0.5em;
  line-height: 1.3;
}

h1, h2, h3 {
  border-bottom: 2px solid silver;
}
h2 {
  padding-top: 0.5em;
}
h3 {
  float: left;
}
h3 + * {
  clear: left;
}
h5 {
  font-size: 1.0em;
}

div.sectionbody {
  margin-left: 0;
}

hr {
  border: 1px solid silver;
}

p {
  margin-top: 0.5em;
  margin-bottom: 0.5em;
}

ul, ol, li > p {
  margin-top: 0;
}
ul > li     { color: #aaa; }
ul > li > * { color: black; }

.monospaced, code, pre {
  font-family: "Courier New", Courier, monospace;
  font-size: inherit;
  color: navy;
  padding: 0;
  margin: 0;
}
pre {
  white-space: pre-wrap;
}

#author {
  color: #527bbd;
  font-weight: bold;
  font-size: 1.1em;
}
#email {
}
#revnumber, #revdate, #revremark {
}

#footer {
  font-size: small;
  border-top: 2px solid silver;
  padding-top: 0.5em;
  margin-top: 4.0em;
}
#footer-text {
  float: left;
  padding-bottom: 0.5em;
}
#footer-badges {
  float: right;
  padding-bottom: 0.5em;
}

#preamble {
  margin-top: 1.5em;
  margin-bottom: 1.5em;
}
div.imageblock, div.exampleblock, div.verseblock,
div.quoteblock, div.literalblock, div.listingblock, div.sidebarblock,
div.admonitionblock {
  margin-top: 1.0em;
  margin-bottom: 1.5em;
}
div.admonitionblock {
  margin-top: 2.0em;
  margin-bottom: 2.0em;
  margin-right: 10%;
  color: #606060;
}

div.content { /* Block element content. */
  padding: 0;
}

/* Block element titles. */
div.title, caption.title {
  color: #527bbd;
  font-weight: bold;
  text-align: left;
  margin-top: 1.0em;
  margin-bottom: 0.5em;
}
div.title + * {
  margin-top: 0;
}

td div.title:first-child {
  margin-top: 0.0em;
}
div.content div.title:first-child {
  margin-top: 0.0em;
}
div.content + div.title {
  margin-top: 0.0em;
}

div.sidebarblock > div.content {
  background: #ffffee;
  border: 1px solid #dddddd;
  border-left: 4px solid #f0f0f0;
  padding: 0.5em;
}

div.listingblock > div.content {
  border: 1px solid #dddddd;
  border-left: 5px solid #f0f0f0;
  background: #f8f8f8;
  padding: 0.5em;
}

div.quoteblock, div.verseblock {
  padding-left: 1.0em;
  margin-left: 1.0em;
  margin-right: 10%;
  border-left: 5px solid #f0f0f0;
  color: #888;
}

div.quoteblock > div.attribution {
  padding-top: 0.5em;
  text-align: right;
}

div.verseblock > pre.content {
  font-family: inherit;
  font-size: inherit;
}
div.verseblock > div.attribution {
  padding-top: 0.75em;
  text-align: left;
}
/* DEPRECATED: Pre version 8.2.7 verse style literal block. */
div.verseblock + div.attribution {
  text-align: left;
}

div.admonitionblock .icon {
  vertical-align: top;
  font-size: 1.1em;
  font-weight: bold;
  text-decoration: underline;
  color: #527bbd;
  padding-right: 0.5em;
}
div.admonitionblock td.content {
  padding-left: 0.5em;
  border-left: 3px solid #dddddd;
}

div.exampleblock > div.content {
  border-left: 3px solid #dddddd;
  padding-left: 0.5em;
}

div.imageblock div.content { padding-left: 0; }
span.image img { border-style: none; vertical-align: text-bottom; }
a.image:visited { color: white; }

dl {
  margin-top: 0.8em;
  margin-bottom: 0.8em;
}
dt {
  margin-top: 0.5em;
  margin-bottom: 0;
  font-style: normal;
  color: navy;
}
dd > *:first-child {
  margin-top: 0.1em;
}

ul, ol {
    list-style-position: outside;
}
ol.arabic {
  list-style-type: decimal;
}
ol.loweralpha {
  list-style-type: lower-alpha;
}
ol.upperalpha {
  list-style-type: upper-alpha;
}
ol.lowerroman {
  list-style-type: lower-roman;
}
ol.upperroman {
  list-style-type: upper-roman;
}

div.compact ul, div.compact ol,
div.compact p, div.compact p,
div.compact div, div.compact div {
  margin-top: 0.1em;
  margin-bottom: 0.1em;
}

tfoot {
  font-weight: bold;
}
td > div.verse {
  white-space: pre;
}

div.hdlist {
  margin-top: 0.8em;
  margin-bottom: 0.8em;
}
div.hdlist tr {
  padding-bottom: 15px;
}
dt.hdlist1.strong, td.hdlist1.strong {
  font-weight: bold;
}
td.hdlist1 {
  vertical-align: top;
  font-style: normal;
  padding-right: 0.8em;
  color: navy;
}
td.hdlist2 {
  vertical-align: top;
}
div.hdlist.compact tr {
  margin: 0;
  padding-bottom: 0;
}

.comment {
  background: yellow;
}

.footnote, .footnoteref {
  font-size: 0.8em;
}

span.footnote, span.footnoteref {
  vertical-align: super;
}

#footnotes {
  margin: 20px 0 20px 0;
  padding: 7px 0 0 0;
}

#footnotes div.footnote {
  margin: 0 0 5px 0;
}

#footnotes hr {
  border: none;
  border-top: 1px solid silver;
  height: 1px;
  text-align: left;
  margin-left: 0;
  width: 20%;
  min-width: 100px;
}

div.colist td {
  padding-right: 0.5em;
  padding-bottom: 0.3em;
  vertical-align: top;
}
div.colist td img {
  margin-top: 0.3em;
}

@media print {
  #footer-badges { display: none; }
}

#toc {
  margin-bottom: 2.5em;
}

#toctitle {
  color: #527bbd;
  font-size: 1.1em;
  font-weight: bold;
  margin-top: 1.0em;
  margin-bottom: 0.1em;
}

div.toclevel0, div.toclevel1, div.toclevel2, div.toclevel3, div.toclevel4 {
  margin-top: 0;
  margin-bottom: 0;
}
div.toclevel2 {
  margin-left: 2em;
  font-size: 0.9em;
}
div.toclevel3 {
  margin-left: 4em;
  font-size: 0.9em;
}
div.toclevel4 {
  margin-left: 6em;
  font-size: 0.9em;
}

span.aqua { color: aqua; }
span.black { color: black; }
span.blue { color: blue; }
span.fuchsia { color: fuchsia; }
span.gray { color: gray; }
span.green { color: green; }
span.lime { color: lime; }
span.maroon { color: maroon; }
span.navy { color: navy; }
span.olive { color: olive; }
span.purple { color: purple; }
span.red { color: red; }
span.silver { color: silver; }
span.teal { color: teal; }
span.white { color: white; }
span.yellow { color: yellow; }

span.aqua-background { background: aqua; }
span.black-background { background: black; }
span.blue-background { background: blue; }
span.fuchsia-background { background: fuchsia; }
span.gray-background { background: gray; }
span.green-background { background: green; }
span.lime-background { background: lime; }
span.maroon-background { background: maroon; }
span.navy-background { background: navy; }
span.olive-background { background: olive; }
span.purple-background { background: purple; }
span.red-background { background: red; }
span.silver-background { background: silver; }
span.teal-background { background: teal; }
span.white-background { background: white; }
span.yellow-background { background: yellow; }

span.big { font-size: 2em; }
span.small { font-size: 0.6em; }

span.underline { text-decoration: underline; }
span.overline { text-decoration: overline; }
span.line-through { text-decoration: line-through; }

div.unbreakable { page-break-inside: avoid; }


/*
 * xhtml11 specific
 *
 * */

div.tableblock {
  margin-top: 1.0em;
  margin-bottom: 1.5em;
}
div.tableblock > table {
  border: 3px solid #527bbd;
}
thead, p.table.header {
  font-weight: bold;
  color: #527bbd;
}
p.table {
  margin-top: 0;
}
/* Because the table frame attribute is overriden by CSS in most browsers. */
div.tableblock > table[frame="void"] {
  border-style: none;
}
div.tableblock > table[frame="hsides"] {
  border-left-style: none;
  border-right-style: none;
}
div.tableblock > table[frame="vsides"] {
  border-top-style: none;
  border-bottom-style: none;
}


/*
 * html5 specific
 *
 * */

table.tableblock {
  margin-top: 1.0em;
  margin-bottom: 1.5em;
}
thead, p.tableblock.header {
  font-weight: bold;
  color: #527bbd;
}
p.tableblock {
  margin-top: 0;
}
table.tableblock {
  border-width: 3px;
  border-spacing: 0px;
  border-style: solid;
  border-color: #527bbd;
  border-collapse: collapse;
}
th.tableblock, td.tableblock {
  border-width: 1px;
  padding: 4px;
  border-style: solid;
  border-color: #527bbd;
}

table.tableblock.frame-topbot {
  border-left-style: hidden;
  border-right-style: hidden;
}
table.tableblock.frame-sides {
  border-top-style: hidden;
  border-bottom-style: hidden;
}
table.tableblock.frame-none {
  border-style: hidden;
}

th.tableblock.halign-left, td.tableblock.halign-left {
  text-align: left;
}
th.tableblock.halign-center, td.tableblock.halign-center {
  text-align: center;
}
th.tableblock.halign-right, td.tableblock.halign-right {
  text-align: right;
}

th.tableblock.valign-top, td.tableblock.valign-top {
  vertical-align: top;
}
th.tableblock.valign-middle, td.tableblock.valign-middle {
  vertical-align: middle;
}
th.tableblock.valign-bottom, td.tableblock.valign-bottom {
  vertical-align: bottom;
}


/*
 * manpage specific
 *
 * */

body.manpage h1 {
  padding-top: 0.5em;
  padding-bottom: 0.5em;
  border-top: 2px solid silver;
  border-bottom: 2px solid silver;
}
body.manpage h2 {
  border-style: none;
}
body.manpage div.sectionbody {
  margin-left: 3em;
}

@media print {
  body.manpage div#toc { display: none; }
}


</style>
<script type="text/javascript">
/*<![CDATA[*/
var asciidoc = {  // Namespace.

/////////////////////////////////////////////////////////////////////
// Table Of Contents generator
/////////////////////////////////////////////////////////////////////

/* Author: Mihai Bazon, September 2002
 * http://students.infoiasi.ro/~mishoo
 *
 * Table Of Content generator
 * Version: 0.4
 *
 * Feel free to use this script under the terms of the GNU General Public
 * License, as long as you do not remove or alter this notice.
 */

 /* modified by Troy D. Hanson, September 2006. License: GPL */
 /* modified by Stuart Rackham, 2006, 2009. License: GPL */

// toclevels = 1..4.
toc: function (toclevels) {

  function getText(el) {
    var text = "";
    for (var i = el.firstChild; i != null; i = i.nextSibling) {
      if (i.nodeType == 3 /* Node.TEXT_NODE */) // IE doesn't speak constants.
        text += i.data;
      else if (i.firstChild != null)
        text += getText(i);
    }
    return text;
  }

  function TocEntry(el, text, toclevel) {
    this.element = el;
    this.text = text;
    this.toclevel = toclevel;
  }

  function tocEntries(el, toclevels) {
    var result = new Array;
    var re = new RegExp('[hH]([1-'+(toclevels+1)+'])');
    // Function that scans the DOM tree for header elements (the DOM2
    // nodeIterator API would be a better technique but not supported by all
    // browsers).
    var iterate = function (el) {
      for (var i = el.firstChild; i != null; i = i.nextSibling) {
        if (i.nodeType == 1 /* Node.ELEMENT_NODE */) {
          var mo = re.exec(i.tagName);
          if (mo && (i.getAttribute("class") || i.getAttribute("className")) != "float") {
            result[result.length] = new TocEntry(i, getText(i), mo[1]-1);
          }
          iterate(i);
        }
      }
    }
    iterate(el);
    return result;
  }

  var toc = document.getElementById("toc");
  if (!toc) {
    return;
  }

  // Delete existing TOC entries in case we're reloading the TOC.
  var tocEntriesToRemove = [];
  var i;
  for (i = 0; i < toc.childNodes.length; i++) {
    var entry = toc.childNodes[i];
    if (entry.nodeName.toLowerCase() == 'div'
     && entry.getAttribute("class")
     && entry.getAttribute("class").match(/^toclevel/))
      tocEntriesToRemove.push(entry);
  }
  for (i = 0; i < tocEntriesToRemove.length; i++) {
    toc.removeChild(tocEntriesToRemove[i]);
  }

  // Rebuild TOC entries.
  var entries = tocEntries(document.getElementById("content"), toclevels);
  for (var i = 0; i < entries.length; ++i) {
    var entry = entries[i];
    if (entry.element.id == "")
      entry.element.id = "_toc_" + i;
    var a = document.createElement("a");
    a.href = "#" + entry.element.id;
    a.appendChild(document.createTextNode(entry.text));
    var div = document.createElement("div");
    div.appendChild(a);
    div.className = "toclevel" + entry.toclevel;
    toc.appendChild(div);
  }
  if (entries.length == 0)
    toc.parentNode.removeChild(toc);
},


/////////////////////////////////////////////////////////////////////
// Footnotes generator
/////////////////////////////////////////////////////////////////////

/* Based on footnote generation code from:
 * http://www.brandspankingnew.net/archive/2005/07/format_footnote.html
 */

footnotes: function () {
  // Delete existing footnote entries in case we're reloading the footnodes.
  var i;
  var noteholder = document.getElementById("footnotes");
  if (!noteholder) {
    return;
  }
  var entriesToRemove = [];
  for (i = 0; i < noteholder.childNodes.length; i++) {
    var entry = noteholder.childNodes[i];
    if (entry.nodeName.toLowerCase() == 'div' && entry.getAttribute("class") == "footnote")
      entriesToRemove.push(entry);
  }
  for (i = 0; i < entriesToRemove.length; i++) {
    noteholder.removeChild(entriesToRemove[i]);
  }

  // Rebuild footnote entries.
  var cont = document.getElementById("content");
  var spans = cont.getElementsByTagName("span");
  var refs = {};
  var n = 0;
  for (i=0; i<spans.length; i++) {
    if (spans[i].className == "footnote") {
      n++;
      var note = spans[i].getAttribute("data-note");
      if (!note) {
        // Use [\s\S] in place of . so multi-line matches work.
        // Because JavaScript has no s (dotall) regex flag.
        note = spans[i].innerHTML.match(/\s*\[([\s\S]*)]\s*/)[1];
        spans[i].innerHTML =
          "[<a id='_footnoteref_" + n + "' href='#_footnote_" + n +
          "' title='View footnote' class='footnote'>" + n + "</a>]";
        spans[i].setAttribute("data-note", note);
      }
      noteholder.innerHTML +=
        "<div class='footnote' id='_footnote_" + n + "'>" +
        "<a href='#_footnoteref_" + n + "' title='Return to text'>" +
        n + "</a>. " + note + "</div>";
      var id =spans[i].getAttribute("id");
      if (id != null) refs["#"+id] = n;
    }
  }
  if (n == 0)
    noteholder.parentNode.removeChild(noteholder);
  else {
    // Process footnoterefs.
    for (i=0; i<spans.length; i++) {
      if (spans[i].className == "footnoteref") {
        var href = spans[i].getElementsByTagName("a")[0].getAttribute("href");
        href = href.match(/#.*/)[0];  // Because IE return full URL.
        n = refs[href];
        spans[i].innerHTML =
          "[<a href='#_footnote_" + n +
          "' title='View footnote' class='footnote'>" + n + "</a>]";
      }
    }
  }
},

install: function(toclevels) {
  var timerId;

  function reinstall() {
    asciidoc.footnotes();
    if (toclevels) {
      asciidoc.toc(toclevels);
    }
  }

  function reinstallAndRemoveTimer() {
    clearInterval(timerId);
    reinstall();
  }

  timerId = setInterval(reinstall, 500);
  if (document.addEventListener)
    document.addEventListener("DOMContentLoaded", reinstallAndRemoveTimer, false);
  else
    window.onload = reinstallAndRemoveTimer;
}

}
asciidoc.install();
/*]]>*/
</script>
</head>
<body class="article">
<div id="header">
<h1>TransDecoder (Find Coding Regions Within Transcripts)</h1>
</div>
<div id="content">
<div id="preamble">
<div class="sectionbody">
<div class="paragraph"><p>TransDecoder identifies candidate coding regions within transcript sequences, such as those generated by de novo RNA-Seq transcript assembly using Trinity, or constructed based on RNA-Seq alignments to the genome using Tophat and Cufflinks.</p></div>
<div class="paragraph"><p>TransDecoder identifies likely coding sequences based on the following criteria:</p></div>
<div class="ulist"><ul>
<li>
<p>
a minimum length open reading frame (ORF) is found in a transcript sequence
</p>
</li>
<li>
<p>
a log-likelihood score similar to what is computed by the GeneID software is &gt; 0.
</p>
</li>
<li>
<p>
the above coding score is greatest when the ORF is scored in the 1st reading frame as compared to scores in the other 5 reading frames.
</p>
</li>
<li>
<p>
if a candidate ORF is found fully encapsulated by the coordinates of another candidate ORF, the longer one is reported.  However, a single transcript can report multiple ORFs (allowing for operons, chimeras, etc).
</p>
</li>
<li>
<p>
<strong>optional</strong> the putative peptide has a match to a Pfam domain above the noise cutoff score.
</p>
</li>
</ul></div>
<div class="paragraph"><p>The software is primarily maintained by <a href="http://www.broadinstitute.org/~bhaas/">Brian Haas</a> at the <a href="http://broadinstitute.org">Broad Institute</a> and <a href="http://tiny.cc/alexie_pap_csiro/">Alexie Papanicolaou</a> at the <a href="http://www.csiro.au/">Commonwealth Scientific and Industrial Research Organisation</a> (CSIRO). It is integrated into other related software such as <a href="http://trinityrnaseq.sf.net">Trinity</a>, <a href="http://pasa.sf.net">PASA</a>, <a href="http://evidencemodeler.sf.net">EVidenceModeler</a>, and <a href="http://trinotate.sf.net">Trinotate</a>.</p></div>
</div>
</div>
<div class="sect1">
<h2 id="_obtaining_transdecoder">Obtaining TransDecoder</h2>
<div class="sectionbody">
<div class="paragraph"><p>The latest release of TransDecoder can be found <a href="https://github.com/TransDecoder/TransDecoder/releases">here</a>.</p></div>
<div class="paragraph"><p>Build it by typing <em>make</em> in the base installation directory (note, the only software compiled is <a href="http://weizhongli-lab.org/cd-hit/">cd-hit</a>, which comes bundled with and is used by TransDecoder).</p></div>
</div>
</div>
<div class="sect1">
<h2 id="_running_transdecoder">Running TransDecoder</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_predicting_coding_regions_from_a_transcript_fasta_file">Predicting coding regions from a transcript fasta file</h3>
<div class="paragraph"><p>The <em>TransDecoder</em> utility is run on a fasta file containing the target transcript sequences.  The simplest usage is as follows:</p></div>
<div class="listingblock">
<div class="content"><!-- Generator: GNU source-highlight 3.1.7
by Lorenzo Bettini
http://www.lorenzobettini.it
http://www.gnu.org/software/src-highlite -->
<pre><tt><span style="font-style: italic"><span style="color: #9A1900"># Step 1: extract the long open reading frames</span></span>
TransDecoder<span style="color: #990000">.</span>LongOrfs -t target_transcripts<span style="color: #990000">.</span>fasta</tt></pre></div></div>
<div class="admonitionblock">
<table><tr>
<td class="icon">
<div class="title">Note</div>
</td>
<td class="content">By default, TransDecoder.LongOrfs will identify ORFs that are at least 100 amino acids long. You can lower this via the <em>-m</em> parameter, but know that the rate of false positive ORF predictions increases drastically with shorter minimum length criteria.</td>
</tr></table>
</div>
<div class="listingblock">
<div class="content"><!-- Generator: GNU source-highlight 3.1.7
by Lorenzo Bettini
http://www.lorenzobettini.it
http://www.gnu.org/software/src-highlite -->
<pre><tt><span style="font-style: italic"><span style="color: #9A1900"># Step 2: (optional)</span></span>
Optionally<span style="color: #990000">,</span> identify ORFs with homology to known proteins via blast or pfam searches<span style="color: #990000">.</span></tt></pre></div></div>
<div class="paragraph"><p>See <a href="#incl_homology">Including homology searches as ORF retention criteria</a> section below.</p></div>
<div class="listingblock">
<div class="content"><!-- Generator: GNU source-highlight 3.1.7
by Lorenzo Bettini
http://www.lorenzobettini.it
http://www.gnu.org/software/src-highlite -->
<pre><tt><span style="font-style: italic"><span style="color: #9A1900"># Step 3: predict the likely coding regions</span></span>
TransDecoder<span style="color: #990000">.</span>Predict -t target_transcripts<span style="color: #990000">.</span>fasta <span style="color: #990000">[</span> homology options <span style="color: #990000">]</span></tt></pre></div></div>
<div class="paragraph"><p>If the transcripts are oriented according to the sense strand, then include the -S flag to examine only the top strand.  Full usage info is below.</p></div>
<div class="paragraph"><p>The final set of candidate coding regions can be found as files <em><strong>.transdecoder.</strong></em> where extensions include .pep, .cds, .gff3, and .bed</p></div>
</div>
<div class="sect2">
<h3 id="_starting_from_a_genome_based_transcript_structure_gtf_file_eg_cufflinks">Starting from a genome-based transcript structure GTF file (eg. cufflinks)</h3>
<div class="paragraph"><p>The process here is identical to the above with the exception that we must first generate a fasta file corresponding to the transcript sequences, and in the end, we recompute a genome annotation file in GFF3 format that describes the predicted coding regions in the context of the genome.</p></div>
<div class="paragraph"><p>Construct the transcript fasta file using the genome and the transcripts.gtf file like so:</p></div>
<div class="listingblock">
<div class="content"><!-- Generator: GNU source-highlight 3.1.7
by Lorenzo Bettini
http://www.lorenzobettini.it
http://www.gnu.org/software/src-highlite -->
<pre><tt>util/cufflinks_gtf_genome_to_cdna_fasta<span style="color: #990000">.</span>pl transcripts<span style="color: #990000">.</span>gtf <span style="font-weight: bold"><span style="color: #0000FF">test</span></span><span style="color: #990000">.</span>genome<span style="color: #990000">.</span>fasta <span style="color: #990000">&gt;</span> transcripts<span style="color: #990000">.</span>fasta</tt></pre></div></div>
<div class="paragraph"><p>Next, convert the transcript structure GTF file to an alignment-GFF3 formatted file (this is done only because our processes operate on gff3 rather than the starting gtf file - nothing of great consequence).  Convert gtf to alignment-gff3 like so, using cufflinks GTF output as an example:</p></div>
<div class="listingblock">
<div class="content"><!-- Generator: GNU source-highlight 3.1.7
by Lorenzo Bettini
http://www.lorenzobettini.it
http://www.gnu.org/software/src-highlite -->
<pre><tt>util/cufflinks_gtf_to_alignment_gff3<span style="color: #990000">.</span>pl transcripts<span style="color: #990000">.</span>gtf <span style="color: #990000">&gt;</span> transcripts<span style="color: #990000">.</span>gff3</tt></pre></div></div>
<div class="paragraph"><p>Now, run the process described above to generate your best candidate ORF predictions:</p></div>
<div class="listingblock">
<div class="content"><!-- Generator: GNU source-highlight 3.1.7
by Lorenzo Bettini
http://www.lorenzobettini.it
http://www.gnu.org/software/src-highlite -->
<pre><tt>TransDecoder<span style="color: #990000">.</span>LongOrfs -t transcripts<span style="color: #990000">.</span>fasta
<span style="color: #990000">(</span>optionally<span style="color: #990000">,</span> identify peptides with homology to known proteins<span style="color: #990000">)</span>
TransDecoder<span style="color: #990000">.</span>Predict -t transcripts<span style="color: #990000">.</span>fasta <span style="color: #990000">[</span> homology options <span style="color: #990000">]</span></tt></pre></div></div>
<div class="paragraph"><p>And finally, generate a genome-based coding region annotation file:</p></div>
<div class="listingblock">
<div class="content"><!-- Generator: GNU source-highlight 3.1.7
by Lorenzo Bettini
http://www.lorenzobettini.it
http://www.gnu.org/software/src-highlite -->
<pre><tt>util/cdna_alignment_orf_to_genome_orf<span style="color: #990000">.</span>pl transcripts<span style="color: #990000">.</span>fasta<span style="color: #990000">.</span>transdecoder<span style="color: #990000">.</span>gff3 transcripts<span style="color: #990000">.</span>gff3 transcripts<span style="color: #990000">.</span>fasta <span style="color: #990000">&gt;</span> transcripts<span style="color: #990000">.</span>fasta<span style="color: #990000">.</span>transdecoder<span style="color: #990000">.</span>genome<span style="color: #990000">.</span>gff3</tt></pre></div></div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_sample_data_and_execution">Sample data and execution</h2>
<div class="sectionbody">
<div class="paragraph"><p>The sample_data/ directory includes a <em>runMe.sh</em> script that you can execute to demonstrate the entire process, starting from a cufflinks GTF file. Note, the typical use-case for TransDecoder is starting from a fasta file containing target <em>Transcripts</em>, however, in the case of genome analysis, transcripts are often inferred from annotation coordinates, such as in this Cufflinks GTF formatted file.  In this example, transcript sequences are reconstructed based on the GTF annotation coordinates, and then TransDecoder is executed on that fasta file.  We include an additional utility for converting the transcript ORF coordinates into genome-coordinates so these regions can be examined in the genomic context.</p></div>
</div>
</div>
<div class="sect1">
<h2 id="_output_files_explained">Output files explained</h2>
<div class="sectionbody">
<div class="paragraph"><p>A working directory (ex. transcripts.transdecoder_dir/) is created to run and store intermediate parts of the pipeline, and contains:</p></div>
<div class="literalblock">
<div class="content">
<pre><code>longest_orfs.pep   : all ORFs meeting the minimum length criteria, regardless of coding potential.
longest_orfs.gff3  : positions of all ORFs as found in the target transcripts
longest_orfs.cds   : the nucleotide coding sequence for all detected ORFs</code></pre>
</div></div>
<div class="literalblock">
<div class="content">
<pre><code>longest_orfs.cds.top_500_longest   : the top 500 longest ORFs, used for training a Markov model for coding sequences.</code></pre>
</div></div>
<div class="literalblock">
<div class="content">
<pre><code>hexamer.scores                     : log likelihood score for each k-mer  (coding/random)</code></pre>
</div></div>
<div class="literalblock">
<div class="content">
<pre><code>longest_orfs.cds.scores            : the log likelihood sum scores for each ORF across each of the 6 reading frames
longest_orfs.cds.scores.selected   : the accessions of the ORFs that were selected based on the scoring criteria (described at top)
longest_orfs.cds.best_candidates.gff3               : the positions of the selected ORFs in transcripts</code></pre>
</div></div>
<div class="paragraph"><p>Then, the final outputs are reported in your current working directory:</p></div>
<div class="literalblock">
<div class="content">
<pre><code>transcripts.fasta.transdecoder.pep : peptide sequences for the final candidate ORFs; all shorter candidates within longer ORFs were removed.
transcripts.fasta.transdecoder.cds  : nucleotide sequences for coding regions of the final candidate ORFs
transcripts.fasta.transdecoder.gff3 : positions within the target transcripts of the final selected ORFs
transcripts.fasta.transdecoder.bed  : bed-formatted file describing ORF positions, best for viewing using GenomeView or IGV.</code></pre>
</div></div>
</div>
</div>
<div class="sect1">
<h2 id="incl_homology">Including homology searches as ORF retention criteria</h2>
<div class="sectionbody">
<div class="paragraph"><p>To further maximize sensitivity for capturing ORFs that may have functional significance, regardless of coding likelihood score as mentioned above, you can scan all ORFs for homology to known proteins and retain all such ORFs.  This can be done in two popular ways: a BLAST search against a database of known proteins, and searching PFAM to identify common protein domains. In the context of TransDecoder, this is done as follows:</p></div>
<div class="paragraph"><p></p></div>
<div class="sect2">
<h3 id="_blastp_search">BlastP Search</h3>
<div class="paragraph"><p>Search a protein database such as <a href="http://www.uniprot.org/">Swissprot (fast) or Uniref90 (slow but more comprehensive)</a> using <a href="http://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE_TYPE=BlastDocs&amp;DOC_TYPE=Download">BLAST+</a>.</p></div>
<div class="paragraph"><p>An example command would be like so:</p></div>
<div class="listingblock">
<div class="content"><!-- Generator: GNU source-highlight 3.1.7
by Lorenzo Bettini
http://www.lorenzobettini.it
http://www.gnu.org/software/src-highlite -->
<pre><tt>blastp -query transdecoder_dir/longest_orfs<span style="color: #990000">.</span>pep  -db uniprot_sprot<span style="color: #990000">.</span>fasta  -max_target_seqs <span style="color: #993399">1</span> -outfmt <span style="color: #993399">6</span> -evalue <span style="color: #993399">1e-5</span> -num_threads <span style="color: #993399">10</span> <span style="color: #990000">&gt;</span> blastp<span style="color: #990000">.</span>outfmt6</tt></pre></div></div>
<div class="paragraph"><p>If you have access to a computing grid, consider using <a href="https://github.com/HpcGridRunner/HpcGridRunner.github.io/releases">HPC GridRunner</a> for more efficient parallel computing.</p></div>
</div>
<div class="sect2">
<h3 id="_pfam_search">Pfam Search</h3>
<div class="paragraph"><p>Search the peptides for protein domains using Pfam. This requires <a href="http://hmmer.janelia.org/">hmmer3</a> and <a href="ftp://ftp.ebi.ac.uk/pub/databases/Pfam/current_release/Pfam-A.hmm.gz">Pfam</a> databases to be installed.</p></div>
<div class="listingblock">
<div class="content"><!-- Generator: GNU source-highlight 3.1.7
by Lorenzo Bettini
http://www.lorenzobettini.it
http://www.gnu.org/software/src-highlite -->
<pre><tt>hmmscan --cpu <span style="color: #993399">8</span> --domtblout pfam<span style="color: #990000">.</span>domtblout /path/to/Pfam-A<span style="color: #990000">.</span>hmm transdecoder_dir/longest_orfs<span style="color: #990000">.</span>pep</tt></pre></div></div>
<div class="paragraph"><p>Just as with the blast search, if you have access to a computing grid, consider using <a href="https://github.com/HpcGridRunner/HpcGridRunner.github.io/releases">HPC GridRunner</a>.</p></div>
</div>
<div class="sect2">
<h3 id="_integrating_the_blast_and_pfam_search_results_into_coding_region_selection">Integrating the Blast and Pfam search results into coding region selection</h3>
<div class="paragraph"><p>The outputs generated above can be leveraged by TransDecoder to ensure that those peptides with blast hits or domain hits are retained in the set of reported likely coding regions.  Run TransDecoder.Predict like so:</p></div>
<div class="listingblock">
<div class="content"><!-- Generator: GNU source-highlight 3.1.7
by Lorenzo Bettini
http://www.lorenzobettini.it
http://www.gnu.org/software/src-highlite -->
<pre><tt>TransDecoder<span style="color: #990000">.</span>Predict -t target_transcripts<span style="color: #990000">.</span>fasta --retain_pfam_hits pfam<span style="color: #990000">.</span>domtblout --retain_blastp_hits blastp<span style="color: #990000">.</span>outfmt6</tt></pre></div></div>
<div class="paragraph"><p>The final coding region predictions will now include both those regions that have sequence characteristics consistent with coding regions in addition to those that have demonstrated blast homology or pfam domain content.</p></div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_viewing_the_orf_predictions_in_a_genome_browser">Viewing the ORF predictions in a genome browser</h2>
<div class="sectionbody">
<div class="paragraph"><p><a href="http://genomeview.org">GenomeView</a> or <a href="http://www.broadinstitute.org/igv/">IGV</a> are  recommended for viewing the candidate ORFs in the context of the genome or the transcriptome.  Examples below show GenomeView in this context.</p></div>
<div class="sect2">
<h3 id="_viewing_orfs_on_target_transcripts">Viewing ORFs on target transcripts</h3>
<div class="listingblock">
<div class="content"><!-- Generator: GNU source-highlight 3.1.7
by Lorenzo Bettini
http://www.lorenzobettini.it
http://www.gnu.org/software/src-highlite -->
<pre><tt>java -jar <span style="color: #009900">$GENOMEVIEW</span>/genomeview<span style="color: #990000">.</span>jar transcripts<span style="color: #990000">.</span>fasta transcripts<span style="color: #990000">.</span>fasta<span style="color: #990000">.</span>transdecoder<span style="color: #990000">.</span>bed</tt></pre></div></div>
<div class="paragraph"><p>If you lack a genome sequence and are working exclusively with the target transcripts, you can load the transcript fasta file and the ORF predictions (bed file) into GenomeView (see below).</p></div>
<div class="paragraph"><p><span class="image">
<img src="images/genomeview_trans.png" alt="Transcript shown with predicted ORF" />
</span></p></div>
</div>
<div class="sect2">
<h3 id="_viewing_orfs_in_the_context_of_the_transcript_structures_on_the_genome">Viewing ORFs in the context of the transcript structures on the genome</h3>
<div class="listingblock">
<div class="content"><!-- Generator: GNU source-highlight 3.1.7
by Lorenzo Bettini
http://www.lorenzobettini.it
http://www.gnu.org/software/src-highlite -->
<pre><tt>java -jar <span style="color: #009900">$GENOMEVIEW</span>/genomeview<span style="color: #990000">.</span>jar <span style="font-weight: bold"><span style="color: #0000FF">test</span></span><span style="color: #990000">.</span>genome<span style="color: #990000">.</span>fasta transcripts<span style="color: #990000">.</span>bed transcripts<span style="color: #990000">.</span>fasta<span style="color: #990000">.</span>transdecoder<span style="color: #990000">.</span>genome<span style="color: #990000">.</span>bed</tt></pre></div></div>
<div class="paragraph"><p>The original cufflinks-based transcript structures are shown in black, and the predicted coding regions are shown in cyan.</p></div>
<div class="paragraph"><p><span class="image">
<img src="images/genomeview_cufflinks.png" alt="Cufflinks trans in GenomeView with predicted ORFs" />
</span></p></div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_technical_support_and_project_announcements">Technical Support and Project Announcements</h2>
<div class="sectionbody">
<div class="paragraph"><p>Join our TransDecoder google group at <a href="https://groups.google.com/forum/#!forum/transdecoder-users">https://groups.google.com/forum/#!forum/transdecoder-users</a></p></div>
</div>
</div>
</div>
<div id="footnotes"><hr /></div>
<div id="footer">
<div id="footer-text">
Last updated 2015-03-24 08:42:50 EDT
</div>
</div>
</body>
</html>
